pkg/kms/localkms/privkey_import.go

```diff
@@ -80,6 +82,34 @@ func (l *LocalKMS) importECDSAKey(privKey *ecdsa.PrivateKey, kt kms.KeyType,
 		return "", nil, fmt.Errorf("import private EC key failed: invalid ECDSA key type")
 	}
 
+	if len(opts) == 0 {
+		var kid string
+
+		// since the public key exists already in decoded form, use the
+		// IEEEP1363 encoding for marhalling and computing the key id as it is
+		// easier to use
+		tmpKt := kms.ECDSAP256TypeIEEEP1363
+
+		switch kt {
+		case kms.ECDSAP256TypeIEEEP1363, kms.ECDSAP256TypeDER:
+			tmpKt = kms.ECDSAP256TypeIEEEP1363
+		case kms.ECDSAP384TypeIEEEP1363, kms.ECDSAP384TypeDER:
+			tmpKt = kms.ECDSAP384TypeIEEEP1363
+		case kms.ECDSAP521TypeIEEEP1363, kms.ECDSAP521TypeDER:
+			tmpKt = kms.ECDSAP521TypeIEEEP1363
+		}
+
+		c := jwkkid.GetCurveByKMSKeyType(tmpKt)
+		pubKeyBytes := elliptic.Marshal(c, privKey.PublicKey.X, privKey.PublicKey.Y)
+
+		kid, err = CreateKID(pubKeyBytes, tmpKt)
+		if err != nil {
+			return "", nil, fmt.Errorf("import private EC  key failed: %w", err)
+		}
+
+		opts = append(opts, kms.WithKeyID(kid))
+	}
+
```

Comment:
this logic is unnecessary since if no KID option is being passed, the function will call `CreateKID()` implicitly.

Reply:
I take away the above statement, CreateKID() was built after ImportPrivKey(), so it's never called, a random KID is generated instead.

This logic is therefore valid for ECDSA keys here and similar CreateKID() calls you added for Ed25519 and BBS keys.